### PR TITLE
Don't allow duplicate selectors in styles

### PR DESCRIPTION
**Pull request by aristath**
## Description

* Removes the `no-duplicate-selectors` exception from stylelint
* Fixes all stylelint reported issues.

Pros:
* No duplicate code
* Smaller footprint for styles :arrow_right: more sustainable

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-list/style.scss [hunk 1]
```diff
@@ -1,10 +1,4 @@
-.block-editor-block-list__layout .block-editor-block-list__block { // Needs specificity to override inherited styles.
-	.reusable-block-edit-panel * {
-		z-index: z-index(".block-editor-block-list__block .reusable-block-edit-panel *");
```
Comment (aristath) [inline packages/block-editor/src/components/block-list/style.scss:3]: Moved to line 18

File: packages/block-editor/src/components/block-list/style.scss [hunk 2]
```diff
@@ -212,93 +292,6 @@
...
-	// This essentially duplicates the mobile styles for the appender component.
-	// It would be nice to be able to use element queries in that component instead https://github.com/tomhodgins/element-queries-spec
-	.block-editor-block-list__layout {
-		.block-editor-default-block-appender .block-editor-inserter {
-			left: auto;
-			right: $grid-unit-10;
-		}
-	}
-}
-
```
Comment (aristath) [inline packages/block-editor/src/components/block-list/style.scss:301]: This section was moved to line 86

File: packages/block-editor/src/components/block-navigation/style.scss [hunk 3]
```diff
@@ -1,10 +1,6 @@
 $tree-border-width: 2px;
 $tree-item-height: 36px;
 
-.block-editor-block-navigation__container {
-	padding: $grid-unit-10 - $border-width;
-}
-
```
Comment (aristath) [inline packages/block-editor/src/components/block-navigation/style.scss:7]: A few lines below this part we set the padding to `padding: $grid-unit - $border-width;` so this rule was always overridden and never applied.

File: packages/block-editor/src/components/link-control/style.scss [hunk 4]
```diff
@@ -166,10 +166,6 @@ $block-editor-link-control-number-of-actions: 1;
 		text-overflow: ellipsis;
 	}
 
-	.block-editor-link-control__search-item-title mark {
-		color: $gray-900;
-	}
-
```
Comment (aristath) [inline packages/block-editor/src/components/link-control/style.scss:172]: A few lines below this part we set the color to `$black` so this rule was always overridden and never applied.

File: packages/components/src/mobile/color-settings/style.native.scss [hunk 5]
```diff
@@ -8,11 +8,6 @@
 	border-color: $gray-70;
 }
 
-.colorIndicator {
-	width: 24px;
-	height: 24px;
-}
-
```
Comment (aristath) [inline packages/components/src/mobile/color-settings/style.native.scss:15]: These rules were duplicated a few lines below

**Review by kevin940726 — APPROVED** [review]
LGTM 👍

**Comment by aristath** [comment]
Thank you for the review @kevin940726 :+1:

**Pull request merged**